### ISSUE TITLE
Publish latest ICS files online

### ISSUE THREAD

**Issue by zeorin** [beyarkay/eskom-calendar#14]
First, thanks for this project, it's a great initiative.

I think it'd be great if the latest ICS files could be published online somewhere (maybe GitHub pages?), deployed automatically with CI when there's a new release, so that they'd always be up to date. Then users could subscribe to them with their calendar apps (most support subscribing to an ICS file ("feed") online), or even with e.g. [Google Calendar](https://support.google.com/calendar/answer/37100?hl=en&co=GENIE.Platform%3DDesktop)

**Comment by beyarkay** [comment]
Hey, thanks for the kind words!

This already happens actually (unless I'm misunderstanding you, in which case please clarify 😁 ). Go to the [`latest` release](https://github.com/beyarkay/eskom-calendar/releases/tag/latest) and then ctrl-f for your area. Copy and paste the link into your calendar subscription and it should all just work.

However, Google calendar only updates [every 12 or 24 hours](https://support.google.com/calendar/thread/1096466?hl=en&msgid=1096537) but I'm told [this script](https://github.com/derekantrican/GAS-ICS-Sync) can fix that (I haven't personally tested it though)

Long term I would like to have a nicer user interface via github pages that will just give user's a magic link that will subscribe them just by clicking on it, but first priority is to get the rest of South Africa's loadshedding areas supported.

Hope that helps, and let me know if you've got any other questions. Otherwise I'll close this issue in a few days time.

**Comment by zeorin** [comment]
Ah! I didn't realize that was an option. I'd assumed each release would have given unique URLs to each ICS file. I'll try that, thanks!

**Comment by beyarkay** [comment]
City power should be up and running actually, as of https://github.com/beyarkay/eskom-calendar/commit/d766ab4f17ba93472b4882f1d83058124471a14b. Here are the links to the calendars:

- [city-power-1](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-1.ics)
- [city-power-2](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-2.ics)
- [city-power-3](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-3.ics)
- [city-power-4](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-4.ics)
- [city-power-5](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-5.ics)
- [city-power-6](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-6.ics)
- [city-power-7](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-7.ics)
- [city-power-8](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-8.ics)
- [city-power-9](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-9.ics)
- [city-power-10](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-10.ics)
- [city-power-11](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-11.ics)
- [city-power-12](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-12.ics)
- [city-power-13](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-13.ics)
- [city-power-14](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-14.ics)
- [city-power-15](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-15.ics)
- [city-power-16](https://github.com/beyarkay/eskom-calendar/releases/download/latest/city-power-16.ics)

**Comment by zeorin** [comment]
Thanks so much. I've added the calendar to Thunderbird, seems to be fine, but hard to be sure atm since there's no loadshedding so the file is nearly blank!

However, there's high risk of loadshedding so I'm sure I'll have an opportunity to test this soon enough. I'll create a new issue if it doesn't work. Thanks again!

**Comment by zeorin** [comment]
Note: the release links redirect to amazon S3 links with expiring credentials in them.

When I added the calendar to Thunderbird it automatically replaced the calendar's URL with the S3 link, which upon expiry of the credentials would cause an error. In the Thunderbird UI it is not possible to change this URL back to the original, neither when adding the calendar nor by editing it once it has been added.

I am trying a workaround where I've edited the URL using the `about:config` settings editor in Thunderbird that given one direct access to the settings values. Search for `calendar.registry.*.uri` in the search bar of the config editor and replace the S3 uri with the original.
